File: leetcode-questions/0884-uncommon-words-from-two-sentences/0884-uncommon-words-from-two-sentences.py
class Solution:
    def helper(self, counter1, counter2):
        uncommon_words = []
        for word, freq in counter1.items():
            if freq == 1 and word not in counter2:
                uncommon_words.append(word)
        return uncommon_words
    
    def uncommonFromSentences(self, s1: str, s2: str) -> List[str]:
        word_counter_s1 = Counter(s1.split(" "))
        word_counter_s2 = Counter(s2.split(" "))
            
        return self.helper(word_counter_s1, word_counter_s2) + self.helper(word_counter_s2, word_counter_s1)
    
    

File: leetcode-questions/0884-uncommon-words-from-two-sentences/0884-uncommon-words-from-two-sentences.js
/**
 * @param {string} s1
 * @param {string} s2
 * @return {string[]}
 */
const helper = (counter1, counter2) => {
    let uncommon_words = [];
    for (const [word, freq] of counter1.entries()) {
        if (freq === 1 && !counter2.get(word)) {
            uncommon_words.push(word)
        }
    }
    
    return uncommon_words;
}
var uncommonFromSentences = function(s1, s2) {
    let counter1 = new Map();
    let counter2 = new Map();
    
    s1.split(" ").map(word => {
        let count = counter1.get(word);
        if (count >= 1) {
            counter1.set(word, count + 1)
        } else {
            counter1.set(word, 1)
        }
    })
    
    s2.split(" ").map(word => {
        let count = counter2.get(word);
        if (count >= 1) {
            counter2.set(word, count + 1)
        } else {
            counter2.set(word, 1)
        }
    })
    
    return [...helper(counter1, counter2), ...helper(counter2, counter1)]
};